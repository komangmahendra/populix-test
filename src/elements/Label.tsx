import React from 'react';
import {StyleSheet, Text, TextProps} from 'react-native';

export type TitleProps = TextProps & {
  size: 'sm' | 'md' | 'lg' | 'xl' | 'header';
};

export const Label = ({children, size, style, ...props}: TitleProps) => {
  const getSize = () => {
    switch (size) {
      case 'sm':
        return 10;
      case 'md':
        return 14;
      case 'lg':
        return 18;
      case 'xl':
        return 32;
      case 'header':
        return 40;
      default:
        return 14;
    }
  };

  const getFontWeight = () => {
    switch (size) {
      case 'sm':
      case 'md':
      case 'lg':
        return '600';
      case 'xl':
        return '800';
      case 'header':
        return '800';
      default:
        return '500';
    }
  };

  return (
    <Text
      {...props}
      style={[
        style,
        {
          ...styles.container,
          fontSize: getSize(),
          fontWeight: getFontWeight(),
        },
      ]}>
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 2,
  },
});
