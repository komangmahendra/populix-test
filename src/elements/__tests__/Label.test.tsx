import React from 'react';
import rendered from 'react-test-renderer';
import {Label} from '../Label';

describe('Test label', () => {
  test('Should match with snapshot when size is sm', () => {
    const tree = rendered.create(<Label size="sm">Small</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should match with snapshot when size is md', () => {
    const tree = rendered.create(<Label size="md">Medium</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should match with snapshot when size is header', () => {
    const tree = rendered.create(<Label size="header">Header</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should match with snapshot when size is larger', () => {
    const tree = rendered.create(<Label size="lg">Large</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should match with snapshot when size is Xtra large', () => {
    const tree = rendered.create(<Label size="xl">Xtra large</Label>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
