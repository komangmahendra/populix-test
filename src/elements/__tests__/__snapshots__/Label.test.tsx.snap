// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test label Should match with snapshot when size is Xtra large 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontSize": 32,
        "fontWeight": "800",
        "marginBottom": 2,
      },
    ]
  }
>
  Xtra large
</Text>
`;

exports[`Test label Should match with snapshot when size is header 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontSize": 40,
        "fontWeight": "800",
        "marginBottom": 2,
      },
    ]
  }
>
  Header
</Text>
`;

exports[`Test label Should match with snapshot when size is larger 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 2,
      },
    ]
  }
>
  Large
</Text>
`;

exports[`Test label Should match with snapshot when size is md 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontSize": 14,
        "fontWeight": "600",
        "marginBottom": 2,
      },
    ]
  }
>
  Medium
</Text>
`;

exports[`Test label Should match with snapshot when size is sm 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontSize": 10,
        "fontWeight": "600",
        "marginBottom": 2,
      },
    ]
  }
>
  Small
</Text>
`;
